### PR TITLE
Essentiality data model

### PR DESCRIPTION
**Pull request by DSuveges**
The details of the data model is under [#2917](https://github.com/opentargets/issues/issues/2917)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
 # Schemas of the OpenTargets Platform data models
 
-This repository contains schema definitions of some of the data models the OpenTargets Platform deals with.
-Either as an input or generates as an output.
+This repository contains schema definitions of some of the data models the OpenTargets Platform deals with. Either as an input or generates as an output.
```
Comment (ireneisdoomed) [inline README.md:5]:
```suggestion
This repository contains schema definitions of some of the data models the Open Targets Platform deals with. Either as an input or generated as an output.
```

File: pydantic_models/essentiality.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+class MutationType(str, Enum):
+    """Mutation class of a target in the essentiality object."""
+    damaging = 'damaging'
+    hotspot = 'hotspot'
+
+class Screen(BaseModel):
+    depmapId: str = Field(description="Screen identifier on DepMap.", examples=['ACH-001029'])
+    cellLineName: str = Field(description="Screened cell-line name.", examples=['CHLA-10'])
+    diseaseFromSource: Optional[str] = Field(description="Disease represented by the applied cell-line.", examples=['Ewing Sarcoma'])
+    diseaseCellLineId: Optional[str] = Field(description="Disease represented by the applied cell-line.", examples=['SIDM01125'], regex=r'SIDM\d+')
```
Comment (ireneisdoomed) [inline pydantic_models/essentiality.py:19]:
```suggestion
    diseaseCellLineId: Optional[str] = Field(description="Sanger cell passport identifier.", examples=['SIDM01125'], regex=r'SIDM\d+')
```

File: pydantic_models/essentiality.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    mutation: Optional[MutationType] = Field(description='Type of the mutation class of the target.', examples=['hotspot'])
+    geneEffect: float = Field(description='Calculated essentiality effect.')
```
Comment (ireneisdoomed) [inline pydantic_models/essentiality.py:21]:
```suggestion
    geneEffect: float = Field(description='Calculated DEMETER2 essentiality effect.')
```

File: pydantic_models/essentiality.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    class Config:
+        extra = Extra.forbid
+
+class Screens(BaseModel):
+    __root__: List[Screen] = Field(unique=True)
+
+class DepMapEssentiality(BaseModel):
+    tissueId: Optional[str] = Field(description="Uberon tissue identifier.", examples=['UBERON_0004535'], regex=r'UBERON_\d+')
+    tissueName: str = Field(description='Tissue name', examamples='liver')
```
Comment (ireneisdoomed) [inline pydantic_models/essentiality.py:32]:
```suggestion
    tissueName: str = Field(description='Tissue name', examples='liver')
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+[tool.poetry]
+name = "json-schema"
+version = "23.06.1"
```
Comment (ireneisdoomed) [inline pyproject.toml:3]: We will be anchoring the version name to our latest release, something that we do in other places already.

File: schemas/gene-essentiality.json [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+          "pattern": "UBERON_\\d+",
+          "examples": [
+            "UBERON_0004535"
+          ],
+          "type": "string"
+        },
+        "tissueName": {
+          "title": "Tissuename",
+          "description": "Tissue name",
+          "examamples": "liver",
```
Comment (ireneisdoomed) [inline schemas/gene-essentiality.json:126]:
```suggestion
          "examples": "liver",
```

File: pydantic_models/essentiality.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    hotspot = 'hotspot'
+
+class Screen(BaseModel):
+    depmapId: str = Field(description="Screen identifier on DepMap.", examples=['ACH-001029'])
+    cellLineName: str = Field(description="Screened cell-line name.", examples=['CHLA-10'])
+    diseaseFromSource: Optional[str] = Field(description="Disease represented by the applied cell-line.", examples=['Ewing Sarcoma'])
+    diseaseCellLineId: Optional[str] = Field(description="Disease represented by the applied cell-line.", examples=['SIDM01125'], regex=r'SIDM\d+')
+    mutation: Optional[MutationType] = Field(description='Type of the mutation class of the target.', examples=['hotspot'])
+    geneEffect: float = Field(description='Calculated essentiality effect.')
+    expression: Optional[float] = Field(description='Expression level of this gene in the screened cell line.')
```
Comment (ireneisdoomed) [inline pydantic_models/essentiality.py:22]:
```suggestion
    expression: Optional[float] = Field(description='Expression level of this gene in the screened cell line (log2(TPM+1)).')
```

File: pydantic_models/essentiality.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    """Mutation class of a target in the essentiality object."""
+    damaging = 'damaging'
+    hotspot = 'hotspot'
+
+class Screen(BaseModel):
+    depmapId: str = Field(description="Screen identifier on DepMap.", examples=['ACH-001029'])
+    cellLineName: str = Field(description="Screened cell-line name.", examples=['CHLA-10'])
+    diseaseFromSource: Optional[str] = Field(description="Disease represented by the applied cell-line.", examples=['Ewing Sarcoma'])
+    diseaseCellLineId: Optional[str] = Field(description="Disease represented by the applied cell-line.", examples=['SIDM01125'], regex=r'SIDM\d+')
+    mutation: Optional[MutationType] = Field(description='Type of the mutation class of the target.', examples=['hotspot'])
```
Comment (ireneisdoomed) [inline pydantic_models/essentiality.py:20]: I read in the issue that you only saw data for these 2 mutation types, but in Depmap they also include "Other non conserving" and "Other conserving" mutations. 
CABYR has some data points that belong to the "other" category https://depmap.org/portal/gene/CABYR?tab=dependency
Are we missing any data?

File: README.md [hunk 9]
```diff
@@ -12,11 +11,10 @@ Either as an input or generates as an output.
...
+- `TEP.json`: description of [Target Enabling Packages](https://platform-docs.opentargets.org/target/chemical-probes-and-teps#target-enabling-packages).
+- `chemical_probes.json`: description of the [chemical probes](https://platform-docs.opentargets.org/target/chemical-probes-and-teps#chemical-probes) data model.
+- `disease_target_evidence.json`: description of disease-target evidence input. Separate schemas are defined for each data sources.
 - `target_safety.json`: Description of [target safety](https://platform-docs.opentargets.org/target/safety).
-- `molecular_interactions.json`: Definition of the [molecular interactions](https://platform-docs.opentargets.org/target/molecular-interactions) input.
+- `molecular_interactions.json`: definition of the [molecular interactions](https://platform-docs.opentargets.org/target/molecular-interactions) input.
 - `molecular_interactions-output.json`: Definition for the molecular interactions dataset generated by the Platform ETL.
-
-##
+- `gene-essentiality.json`: definition of gene essentiality target annotation. Developed based on [DepMap](https://depmap.org/) CRISPR/CAS9 screens.
```
Comment (ireneisdoomed) [inline README.md:20]:
```suggestion
- `gene-essentiality.json`: definition of gene essentiality target annotation. Developed based on [DepMap](https://depmap.org/) CRISPR/CAS9 screens.

## Dev environment set up
```bash
# Clone the project
git clone https://github.com/opentargets/json_schema
cd json_schema

# Install Poetry
curl -sSL https://install.python-poetry.org | python -

# Install the project dependencies
poetry install
poetry shell
```
```

**Review by ireneisdoomed — APPROVED** [review]
Great, I just added minor comments to improve some field descriptions. The data model follows exactly what is discussed in [#2917]. It should also be straightforward to reproduce their widget.

Thank you too for setting up the poetry env.

File: pydantic_models/essentiality.py (hunk 8 above)
Comment (DSuveges) [reply pydantic_models/essentiality.py:20]: Yes, that pieces of information is quite difficult to extract. Not for this release.

File: pyproject.toml (hunk 5 above)
Comment (DSuveges) [reply pyproject.toml:3]: I don't have strong feeling about about fixing versions in this particular case, where the software in secondary to the data.

Comment (ireneisdoomed) [reply pyproject.toml:3]: Ah, I was just describing. No strong feeling against it either

**Pull request merged**